From 05fe24097fa8287bddea3418799e32d949999267 Mon Sep 17 00:00:00 2001
From: "M. Vefa Bicakci" <vefa.bicakci@windriver.com>
Date: Mon, 14 Feb 2022 13:13:09 -0500
Subject: [PATCH] Set up stdatomic.h if necessary

The toolchain used for building libraries in StarlingX (i.e., CentOS 7)
does not provide the stdatomic.h header. Thankfully, the libbnxt_re
package includes a compatibility-oriented version of this header in
rc-compat/v22.

Hence, copy over this header from the rc-compat/v22 directory to the
current rc-compat version sub-directory, but only if required, to let
the build finish successfully.

Signed-off-by: M. Vefa Bicakci <vefa.bicakci@windriver.com>
---
 configure.ac | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/configure.ac b/configure.ac
index 9e4e980886d4..61f2a014eb94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -481,6 +481,22 @@ AC_HEADER_STDC
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
 
+dnl Check if the host's toolchain provides stdatomic.h and use
+dnl our local copy if required.
+tmp_stdatomic=if$$.c
+echo "#include <stdatomic.h>" > ${tmp_stdatomic}
+dnl Need to work around embedded newlines, hence the additional
+dnl level of indirection.
+incpath=$(echo -n __include_path__)
+if echo -- $incpath | grep -q src/rc-compat/ && \
+   ! $CC $CPPFLAGS -E ${tmp_stdatomic} >/dev/null 2>/dev/null && \
+   ! test -f src/rc-compat/$(echo -n rcore_version)/stdatomic.h; then
+	cp -v src/rc-compat/v22/stdatomic.h src/rc-compat/$(echo -n rcore_version)/stdatomic.h
+fi
+rm -f ${tmp_stdatomic}
+unset tmp_stdatomic
+unset incpath
+
 dnl Check for libibverbs device library extension
 dummy=if$$
 cat <<IBV_VERSION > $dummy.c
-- 
2.29.2

